ray_julia_jll/gen/make.jl

```diff
@@ -0,0 +1,168 @@
+    if archive_uri.host != "github.com"
+        throw(ArgumentError("Artifact URL is not for github.com: $(archive_uri)"))
+    end
+
+    m = match(GH_RELEASE_ASSET_PATH_REGEX, archive_uri.path)
+    if m === nothing
+        throw(ArgumentError(
+            "Artifact URL is not a GitHub release asset path: $(archive_uri)"
+        ))
+    end
+
+    # The `ghr` utility uses the local file name for the release asset. In order to have
+    # have the asset match the specified URL we'll temporarily rename the file.
+    org_archive_name = nothing
+    if basename(archive_path) != m[:file_name]
+        org_archive_name = basename(archive_path)
+        archive_path = mv(archive_path, joinpath(dirname(archive_path), m[:file_name]))
+    end
+
+    upload_to_github_release(m[:owner], m[:repo_name], commit, m[:tag], archive_path; kwargs...)
+
+    # Rename the archive back to the original name
+    if org_archive_name !== nothing
+        mv(archive_path, joinpath(dirname(archive_path), org_archive_name))
+    end
+
+    return nothing
+end
+
+
+function upload_to_github_release(owner, repo_name, commit, tag, path; token=ENV["GITHUB_TOKEN"])
+    # Based on: https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/d40ec617d131a1787851559ef1a9f04efce19f90/src/AutoBuild.jl#L487
+    # TODO: Passing in a directory path uploads multiple assets
+    # TODO: Would be nice to perform parallel uploads
+    cmd = ```
+        $(ghr()) \
+        -owner $owner \
+        -repository $repo_name \
+        -commitish $commit \
+        -token $token \
+        -n $tag \
+        -prerelease \
+        $tag $path
+    ```
+
+    run(cmd)
+end
+
+if abspath(PROGRAM_FILE) == @__FILE__
+
+    # Check for this now before we spend any time building the JLL
+    !haskey(ENV, "GITHUB_TOKEN") && error("\"GITHUB_TOKEN\" environment variable required.")
+
+    repo_path = abspath(joinpath(@__DIR__, "..", ".."))
+    pkg_url = remote_url(repo_path)
+
+    # Build JLL
+    @info "Building ray_julia_jll..."
+    Pkg.build("ray_julia_jll"; verbose=true)
+    compiled_dir = joinpath(repo_path, "ray_julia_jll", "deps", "bazel-bin")
+
+    # Limit what we include in tarball
+    compiled_assets = Set(readdir(compiled_dir))
+    if compiled_assets ⊈ ASSETS
+        throw(ArgumentError("Unexpected JLL assets found: $compiled_assets"))
+    end
+
+    # Read Project.toml
+    jll_project_toml = joinpath(repo_path, "ray_julia_jll", "Project.toml")
+    jll_project = read_project(jll_project_toml)
+    jll_version = jll_project.version
+
+    host_triplet = BinaryPlatforms.host_triplet()
+    tarball_name = "ray_julia.v$jll_version.$host_triplet.tar.gz"
+
+    @info "Creating tarball $tarball_name"
+    tarball_path = joinpath(tempdir(), tarball_name)
+    create_tarball(readlink(compiled_dir), tarball_path)
+
+    m = match(URL_REGEX, pkg_url)
+    if m === nothing
+        throw(ArgumentError("Package URL is not a valid SCP or HTTP URL: $(pkg_url)"))
+    end
```

Comment:
Consider making this into a function `parse_git_remote_url(::AbstractString) -> URI`